src/v2/components/ui/BareInput.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useState } from "react";
+import { styled } from "src/v2/stitches.config";
+
+const BareInputWrapper = styled("span", {
+  position: "relative",
+  display: "inline-block",
+});
+
+const LayoutCalculator = styled("span", {
+  visibility: "hidden",
+});
+
+const BareInputStyled = styled("input", {
+  appearance: "none",
+  all: "unset",
+  position: "absolute",
+  width: "100%",
+  left: 0,
+  top: "50%",
+  transform: "translateY(-50%)",
+  "&::-webkit-inner-spin-button": {
+    display: "none",
+  },
+});
+
+function BareInput(
+  props: React.InputHTMLAttributes<HTMLInputElement>,
+  ref: React.Ref<HTMLInputElement>
+) {
+  const [value, setValue] = useState(props.value || "");
+  const onChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      props.onChange?.(e);
+      setValue(e.target.value.slice(0, props.maxLength));
+    },
+    [props.onChange]
+  );
+
+  return (
+    <BareInputWrapper>
+      <LayoutCalculator aria-hidden>{value}</LayoutCalculator>
```

Comment:
Well, this bit is a bit of magic that I 'borrowed' from CSS-Tricks. Basically, this works by letting the browser calculate the width of input by copying the input's value into an inline element, causing the parent to get resized.

This was necessary because the current amount indicator has to come right after the input and the width of both elements are set dynamically.

* https://css-tricks.com/auto-growing-inputs-textareas/